diff --git a/dist/store/useStore.js b/dist/store/useStore.js
--- a/dist/store/useStore.js
+++ b/dist/store/useStore.js
@@ -1,21 +1,23 @@
-import { useState, useEffect } from 'react';
+import { useState, useEffect, useRef } from 'react';
 import isEqual from 'lodash/isEqual.js';
 import { useEvent } from "../util/index.js";
 import { useStoreContext } from "./useStoreContext.js";
 function useStore(key, defaultValue) {
     var _a = useStoreContext(), getItem = _a.getItem, setItem = _a.setItem, subscribe = _a.subscribe;
     var _b = useState(function () { return getItem(key, defaultValue); }), value = _b[0], setValue = _b[1];
+    var valueRef = useRef(value);
+    valueRef.current = value;
     // subscribe to changes on this key, and change the state when they happen
     useEffect(function () {
         var storedValue = getItem(key, defaultValue);
-        if (!isEqual(value, storedValue)) {
+        if (!isEqual(valueRef.current, storedValue)) {
             setValue(storedValue);
         }
         var unsubscribe = subscribe(key, function (newValue) {
             setValue(typeof newValue === 'undefined' ? defaultValue : newValue);
         });
         return function () { return unsubscribe(); };
-    }, [key, subscribe, defaultValue, getItem, value]);
+    }, [key, subscribe, defaultValue, getItem]);
     var set = useEvent(function (valueParam, runtimeDefaultValue) {
         var newValue = typeof valueParam === 'function' ? valueParam(value) : valueParam;
         // we only set the value in the Store;
diff --git a/dist/store/useStore.cjs b/dist/store/useStore.cjs
--- a/dist/store/useStore.cjs
+++ b/dist/store/useStore.cjs
@@ -11,17 +11,19 @@
 function useStore(key, defaultValue) {
     var _a = (0, useStoreContext_1.useStoreContext)(), getItem = _a.getItem, setItem = _a.setItem, subscribe = _a.subscribe;
     var _b = (0, react_1.useState)(function () { return getItem(key, defaultValue); }), value = _b[0], setValue = _b[1];
+    var valueRef = (0, react_1.useRef)(value);
+    valueRef.current = value;
     // subscribe to changes on this key, and change the state when they happen
     (0, react_1.useEffect)(function () {
         var storedValue = getItem(key, defaultValue);
-        if (!(0, isEqual_js_1.default)(value, storedValue)) {
+        if (!(0, isEqual_js_1.default)(valueRef.current, storedValue)) {
             setValue(storedValue);
         }
         var unsubscribe = subscribe(key, function (newValue) {
             setValue(typeof newValue === 'undefined' ? defaultValue : newValue);
         });
         return function () { return unsubscribe(); };
-    }, [key, subscribe, defaultValue, getItem, value]);
+    }, [key, subscribe, defaultValue, getItem]);
     var set = (0, util_1.useEvent)(function (valueParam, runtimeDefaultValue) {
         var newValue = typeof valueParam === 'function' ? valueParam(value) : valueParam;
         // we only set the value in the Store;
